README.md

```diff
@@ -95,22 +95,38 @@ Again, it has not been officially tested on windows, there shouldn't be anything
 
 ### Native Dependencies
 
-The script explicitly requires the following programs:
-
-| Program       | Role In Script                                         |
-| ------------- | ------------------------------------------------------ |
-| Bash          | Execution of script                                    |
-| Curl          | All network requests                                   |
-| file/mimetype | Mimetype generation for extension less files           |
-| find          | To find files and folders for recursive folder uploads |
-| xargs         | For parallel uploading                                 |
-| mkdir         | To create folders                                      |
-| rm            | To remove files and folders                            |
-| grep          | Miscellaneous                                          |
-| sed           | Miscellaneous                                          |
-| diff          | To detect new files in a folder ( only sync.sh )       |
-| ps            | To manage background jobs ( only sync.sh )             |
-| tail          | To show indefinite logs ( only sync.sh )               |
+This repo contains two types of scripts, posix compatible and bash compatible.
+
+#### These programs are required in both bash and posix scripts.
+
+| Program          | Role In Script                                         |
+| ---------------- | ------------------------------------------------------ |
+| curl             | All network requests                                   |
+| file or mimetype | Mimetype generation for extension less files           |
+| find             | To find files and folders for recursive folder uploads |
+| xargs            | For parallel uploading                                 |
+| mkdir            | To create folders                                      |
+| rm               | To remove files and folders                            |
+| grep             | Miscellaneous                                          |
+| sed              | Miscellaneous                                          |
+| mktemp           | To generate temporary files ( optional )               |
+
+#### If BASH is not available or BASH is available but version is less tham 4.x, then below programs are also required:
+
+| Program             | Role In Script                             |
+| ------------------- | ------------------------------------------ |
+| ps                  | For parallel upload progress               |
+| date                | For installation, update and Miscellaneous |
+| cat                 | Miscellaneous                              |
+| sleep               | Self explanatory                           |
+| stty or zsh or tput | To determine column size ( optional )      |
+
+#### These programs are needed for synchronisation script:
```

Comment:
Codacy found an issue: [[no-heading-punctuation] Don’t add a trailing `:` to headings](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)